examples/medplum-provider/src/pages/encounter/EncounterChart.tsx

```diff
@@ -253,6 +256,28 @@ export const EncounterChart = (): JSX.Element => {
     }, SAVE_TIMEOUT_MS);
   };
 
+  const handleDiagnosisChange = (value: CodeableConcept | undefined): void => {
+    setDiagnosis(value ? value : { coding: [] });
+
+    if (!claim) {
+      return;
+    }
+
+    if (saveTimeoutRef.current) {
+      clearTimeout(saveTimeoutRef.current);
+    }
+
+    saveTimeoutRef.current = setTimeout(async () => {
```

Comment:
I'm a little nervous about scattering `setTimeout()` throughout these components.  I'd much rather have a consistent "auto save" system + philosophy.

I'm ok to merge this for now, but let's have an offline discussion about how we can do this more consistently.